financialaid/serializers.py

```diff
@@ -0,0 +1,54 @@
+"""
+Serializers from financialaid
+"""
+import datetime
+from django.shortcuts import get_object_or_404
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+from rest_framework.fields import IntegerField, FloatField, CharField
+
+from courses.models import Program
+from dashboard.models import ProgramEnrollment
+from financialaid.backend import determine_tier_program, determine_auto_approval
+from financialaid.models import FinancialAid
+
+
+class FinancialAidSerializer(serializers.Serializer):
+    """
+    Serializer for Financial Aid objects
+    """
+    original_income = FloatField(min_value=0)
+    original_currency = CharField()
+    program_id = IntegerField()
+
+    def validate(self, data):
+        """
+        Validators for this serializer
+        """
+        data["program"] = get_object_or_404(Program, pk=data["program_id"])
+        if not data["program"].financial_aid_availability:
+            raise ValidationError("Financial aid not available for this program.")
+        if not ProgramEnrollment.objects.filter(program=data["program"], user=self.context["request"].user).exists():
```

Comment:
Actually I thought more about this and I think we should explicitly pass the user username together with the POST request and not pull it from the context.
This has the complication that we need an extra permission class on the view to enforce that the user can change only her data.
This might be easier if we modify the url to include the username.

I am open to discuss this if you do not agree, but my reason is that the serializer should not be tied to a request context.